### PR TITLE
Overlay Improvements

### PR DESCRIPTION
**Pull request by Richardant**
- Added a different color for the last tick number (Fully customizable, White by default)
- Fixed the height for the tick number to be static (so it doesn't go up and down) and it's now customizable (Top, Centered, Bottom)
- Added Bold Font
- Improved the customization for the Attack Bar Height (From -100 to 100, 0 by default)
- Show Tick enabled by default (This shouldn't be false by default since it's the main feature of the plugin)
- Font size by default increased to 18 (So that it matches the icon banner)
- Added an icon banner for the plugin

### CODE REVIEW

**Comment by Richardant** [comment]
Here's how it looks before and after the changes:

Before it followed the prayer, going up and down depending on player height
https://github.com/user-attachments/assets/3cd6d97c-70ab-43ff-8fd7-250a5249fb7e

After the changes, the tick numbers stays in the same position, adjustable in the config (Top, Centered, Bottom)
https://github.com/user-attachments/assets/30184052-e716-42f1-be36-78957b2ae643

**Comment by Richardant** [comment]
Here's how it looks before and after the changes:

Before it followed the prayer, going up and down depending on player height

https://github.com/user-attachments/assets/448ff80f-d246-4956-a1be-1dce271eb09f

**Comment by Richardant** [comment]
After the changes, the tick numbers stays in the same position, adjustable in the config (Top, Centered, Bottom)

https://github.com/user-attachments/assets/555f09e5-4dcb-46c3-bb81-ac77ae2d1a6c

**Comment by Richardant** [comment]
After the changes, the tick numbers stays in the same position, adjustable in the config (Top, Centered, Bottom)

https://github.com/user-attachments/assets/35c33a8f-b597-4341-8310-df3ebcf61b82

**Comment by Richardant** [comment]
And there's a easy fix for the issue #59 but i'm not sure if i should add it since it overlaps hp bar sometimes


https://github.com/user-attachments/assets/6969648c-2264-4ee8-98a3-5a4b969785ee

`final int height = 194 + config.heightOffset();`

**Comment by Richardant** [comment]
@ngraves95 can you take a look? I'm excited to hear your thoughts on these changes. 😅

**Comment by Lexer747** [comment]
This broke the RS Regular font for me when giving it a go. So that would block me merging this personally.

https://github.com/user-attachments/assets/91d4c49b-74cf-43de-8303-4ff9676d62c3

I actually liked the tick being in-line with prayer so for me this isn't a "fix" bug I would like a 4th option to have it follow. This would also block me merging personally but that depends how ngraves95 would feel about it.

The last tick being a different colour is a nice add-on, good job :)

I barely ever play with the bar on so pass on all that code and options. The defaults changing I'm also ok with.

File: src/main/java/com/attacktimer/AttackTimerMetronomeTileOverlay.java
Comment (Lexer747) [inline src/main/java/com/attacktimer/AttackTimerMetronomeTileOverlay.java:115]: This should be a switch statement.

**Comment by Richardant** [comment]
> This broke the RS Regular font for me when giving it a go. So that would block me merging this personally.
Broken in the sense that it no longer follows the prayer anymore?, if that's the case i can add a 4th option for sure! if you mean something different let me know 😅

File: src/main/java/com/attacktimer/AttackTimerMetronomeTileOverlay.java
Comment (Richardant) [reply src/main/java/com/attacktimer/AttackTimerMetronomeTileOverlay.java:115]: I thought the same thing, I'll fix it

**Comment by Lexer747** [comment]
> > This broke the RS Regular font for me when giving it a go. So that would block me merging this personally.
> 
> Broken in the sense that it no longer follows the prayer anymore?, if that's the case i can add a 4th option for sure! if you mean something different let me know 😅

Nah look at this comparison:

Before: 
java_9ungKII5gx


After:
ApplicationFrameHost_nj8kvLEZbI

That's not the same 4. It should be the same font as the chat box.

**Comment by Richardant** [comment]
> That's not the same 4. It should be the same font as the chat box.

True af, i was doing other fixes and figure it out what you mean, i already fix it

**Comment by Richardant** [comment]
It should be good now but if you find something else please lemme know, Thank you for your feedback @Lexer747

**Review by Lexer747 — APPROVED** [review]
Thanks for fixing those things, and thanks for new options, LGTM now I'd be happy with a merge personally.

There's a few formatting problems in `render`, there should always be a new line after a brace for `if` statements. And squashing the commits into a single one is helpful too.

Though it's up to @ngraves95 ultimately if it gets merged.

**Comment by ngraves95** [comment]
I'll try to take a look this weekend - thanks for putting this together and including videos.

File: src/main/java/com/attacktimer/AttackTimerMetronomePlugin.java
Comment (Richardant) [inline src/main/java/com/attacktimer/AttackTimerMetronomePlugin.java:1]: I did a quick fix for nightmare pillars since they aren't working with the plugin, if there's a different way to fix it lmk

**Comment by Richardant** [comment]
> I'll try to take a look this weekend - thanks for putting this together and including videos.

Any update sir? 👀  @ngraves95

File: src/main/java/com/attacktimer/AttackTimerMetronomeConfig.java
Comment (ngraves95) [inline src/main/java/com/attacktimer/AttackTimerMetronomeConfig.java:42]: I'd prefer to keep default settings the same so that users don't have any surprises after this plugin gets updated.

File: src/main/java/com/attacktimer/AttackTimerMetronomeConfig.java
Comment (ngraves95) [inline src/main/java/com/attacktimer/AttackTimerMetronomeConfig.java:71]: Same comment about keeping the same defaults.

File: src/main/java/com/attacktimer/AttackTimerMetronomeConfig.java
Comment (ngraves95) [inline src/main/java/com/attacktimer/AttackTimerMetronomeConfig.java:94]: Let's default this to Color.CYAN. I do like the option to have a different color though.

File: src/main/java/com/attacktimer/AttackTimerMetronomePlugin.java
Comment (ngraves95) [inline src/main/java/com/attacktimer/AttackTimerMetronomePlugin.java:350]: Maybe worth moving the special npcs into its own enum/file.

File: src/main/java/com/attacktimer/AttackTimerMetronomeTileOverlay.java
Comment (ngraves95) [inline src/main/java/com/attacktimer/AttackTimerMetronomeTileOverlay.java:118]: Huh, I guess height offset never worked in the first place. Thanks for fixing it.

File: src/main/java/com/attacktimer/AttackTimerMetronomeConfig.java
Comment (Richardant) [reply src/main/java/com/attacktimer/AttackTimerMetronomeConfig.java:42]: I was thinking about other examples on plugin hub, main feature of the plugin should be ON by default, so new users don't get confused when they install the plugin for the first time and ask themselves "Why after I install the plugin isn't showing the tick counter? Isn't this the plugin for that?"

Lemme know what you think, if you don't agree we can just set it to false, not a biggie

File: src/main/java/com/attacktimer/AttackTimerMetronomeTileOverlay.java
Comment (ngraves95) [inline src/main/java/com/attacktimer/AttackTimerMetronomeTileOverlay.java:103]: We should de-duplicate the text rendering logic; its the same thing repeated for each case.

File: src/main/java/com/attacktimer/AttackTimerMetronomeConfig.java
Comment (Richardant) [reply src/main/java/com/attacktimer/AttackTimerMetronomeConfig.java:71]: Yeah, I can agree with this one, it's just too small on some screens so I decided to set it a little higher ahah, I will revert this

File: src/main/java/com/attacktimer/AttackTimerMetronomeConfig.java
Comment (Richardant) [reply src/main/java/com/attacktimer/AttackTimerMetronomeConfig.java:94]: Okok, I was thinking the same possibility

File: src/main/java/com/attacktimer/AttackTimerMetronomePlugin.java
Comment (Richardant) [reply src/main/java/com/attacktimer/AttackTimerMetronomePlugin.java:350]: Worth for future updates, I think we can leave it like that meanwhile

File: src/main/java/com/attacktimer/AttackTimerMetronomeConfig.java
Comment (ngraves95) [reply src/main/java/com/attacktimer/AttackTimerMetronomeConfig.java:42]: That makes sense. Let's leave this one defaulted to true.

File: icon.png
Comment (ngraves95) [inline icon.png:1]: Why is this file added?

Comment (Richardant) [reply icon.png:1]: This is the icon for the plugin hub panel, so ppl will have an idea of how the plugin looks, right now the plugin doesn't have any icon

File: src/main/java/com/attacktimer/AttackTimerMetronomeTileOverlay.java
Comment (Richardant) [reply src/main/java/com/attacktimer/AttackTimerMetronomeTileOverlay.java]: Fixed and tested, ty <3

File: icon.png
Comment (Richardant) [reply icon.png:1]: Right now it looks like this 
imagen

with the file that i added (icon.png) it will look like this:
imagen

or like this if i update the current icon.png
imagen

which one do you prefer? black background or grey background?

Comment (Richardant) [reply icon.png:1]: @ngraves95

Comment (Richardant) [reply icon.png:1]: i'm still waiting for your review 😓 @ngraves95

**Comment by Richardant** [comment]
@ngraves95

**Comment by Richardant** [comment]
@ngraves95 👀

File: icon.png
Comment (Richardant) [reply icon.png:1]: @ngraves95

Comment (Richardant) [reply icon.png:1]: @ngraves95

Comment (Richardant) [reply icon.png:1]: @ngraves95

**Comment by Richardant** [comment]
@ngraves95

**Comment by Richardant** [comment]
@ngraves95

**Comment by Richardant** [comment]
Hey sir has been a while 😢 @ngraves95

**Comment by Richardant** [comment]
@ngraves95 Hey nick, can you add me as a contributor?

**Comment by Richardant** [comment]
@ngraves95 Hey nick, can you add give me collaborator access? So i can help you with the repo

**Comment by Richardant** [comment]
@ngraves95 Hey nick, can you give me collaborator access? So i can help you with the repo

**Comment by Richardant** [comment]
ty <3

File: icon.png
Comment (Richardant) [reply icon.png:1]: Aw sir, i was waiting for you opinion in here, grey bg or black bg? @ngraves95

**Comment by Richardant** [comment]
If you are busy, send a message in here indicating that this PR and #66 are OK to update, so I can do it myself @ngraves95

**Comment by Richardant** [comment]
If you are busy, send a message in here indicating that this PR and https://github.com/ngraves95/attacktimer/pull/66 are OK to update, so I can do it myself @ngraves95

**Comment by Richardant** [comment]
If you are busy, send a message in here indicating that this PR and https://github.com/ngraves95/attacktimer/pull/66 are OK to update, so I can do it myself @ngraves95

**Pull request merged**
